packages/typespec-ts/test/modularIntegration/generated/lro_modular/generated/src/common/lroImpl.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  LongRunningOperation,
+  LroResponse,
+  OperationState,
+  SimplePollerLike,
+  createHttpPoller,
+} from "@azure/core-lro";
+import {
+  Client,
+  HttpResponse,
+  StreamableMethod,
+} from "@azure-rest/core-client";
+export interface GetLongRunningPollerOptions<
+  TResponse extends HttpResponse = HttpResponse
+> {
+  requestMethod: string;
+  requestUrl: string;
+  sendInitialRequestFn: (...args: unknown[]) => StreamableMethod<TResponse>;
+  sendInitialRequestFnArgs: unknown[];
+  deserializeFn: (response: TResponse) => Promise<unknown>;
+}
+export async function getLongRunningPoller<TResponse extends HttpResponse>(
+  client: Client,
+  options: GetLongRunningPollerOptions<TResponse>
+): Promise<SimplePollerLike<OperationState<unknown>, unknown>> {
+  let initialResponse: TResponse;
+  const poller: LongRunningOperation<unknown> = {
+    requestMethod: options.requestMethod,
+    requestPath: options.requestUrl,
+    sendInitialRequest: async () => {
+      initialResponse = (await options.sendInitialRequestFn(
+        ...options.sendInitialRequestFnArgs
+      )) as TResponse;
+      return getLroResponse(initialResponse, options.deserializeFn);
+    },
+    sendPollRequest: async (path) => {
+      // This is the callback that is going to be called to poll the service
+      // to get the latest status. We use the client provided and the polling path
+      // which is an opaque URL provided by caller, the service sends this in one of the following headers: operation-location, azure-asyncoperation or location
+      // depending on the lro pattern that the service implements. If non is provided we default to the initial path.
+      if (!path || path === options.requestUrl) {
+        path = initialResponse.request.url ?? options.requestUrl;
+      }
+      const response = await client.pathUnchecked(path).get();
+      response.headers["x-ms-original-url"] = initialResponse.request.url;
+      const lroResponse = getLroResponse(
+        response as TResponse,
+        options.deserializeFn
+      );
+      return lroResponse;
+    },
+  };
+  return await createHttpPoller(poller);
+}
```

Comment:
what's the difference between this one and the one in rlc ? do we really allow customer to use this operation to create poller by themselves ? I don't think we do in current HLC ?

Reply:
> what's the difference between this one and the one in rlc ?
The main difference is the request sequence:
- For RLC we have to send the initial request first and then until the response returned we could proceed to call our LRO helper 
- For Modular we swap all these details and we directly return the poller


> do we really allow customer to use this operation to create poller by themselves ?

I don't think we would open it to customer at the beginning.